utilities/cli/util.py

```diff
@@ -641,3 +650,142 @@ def get_devcontainer_config(
+    finally:
+        try:
+            os.chdir(original_cwd)  # try to restore the original working directory
+        except Exception:
+            pass
+
+
+def collect_docker_info() -> None:
+    """Collect and display Docker information"""
+    print(f"\n{Color.blue('Docker Information:')}")
+    docker_version = run_info_command(["docker", "--version"])
+    docker_info = run_info_command(["docker", "info", "--format", "{{.ServerVersion}}"])
+    if docker_version is None or docker_info is None:
+        print("  Docker not available")
+        return
+    print(f"  Version: {docker_version} Server Version: {docker_info}")
+
+
+def collect_cuda_gpu_info() -> None:
+    """Collect and display CUDA/GPU information"""
+    print(f"\n{Color.blue('CUDA/GPU Information:')}")
+    nvidia_smi = run_info_command(
+        [
+            "nvidia-smi",
+            "--query-gpu=name,driver_version,memory.total",
+            "--format=csv,noheader,nounits",
+        ]
+    )
+    if nvidia_smi is None:
+        print("  NVIDIA GPU/CUDA not available")
+        return
+    for i, line in enumerate(nvidia_smi.split("\n")):
+        if line.strip():
+            parts = line.split(",")
+            if len(parts) >= 3:
+                print(f"  GPU {i}: {parts[0].strip()}")
+                print(f"    Driver: {parts[1].strip()}")
+                print(f"    Memory: {parts[2].strip()} MB")
+    cuda_version = run_info_command(["nvcc", "--version"])
+    if cuda_version is None:
+        print("  nvcc not available")
+        return
+    version_line = [line for line in cuda_version.split("\n") if "release" in line.lower()]
+    print(f"  NVCC: {version_line[0].strip()}")
+    nvcc_path = run_info_command(["which", "nvcc"])
+    if nvcc_path is not None:
+        print(f"  NVCC Path: {nvcc_path}")
+
+
+def collect_environment_variables() -> None:
+    """Collect and display environment variables"""
+    print(f"\n{Color.blue('HoloHub Environment Variables:')}")
+    holohub_env_vars = [
+        "HOLOHUB_CMD_NAME",
+        "HOLOHUB_BUILD_LOCAL",
+        "HOLOHUB_BASE_IMAGE",
+        "HOLOHUB_APP_NAME",
+        "HOLOHUB_CONTAINER_BASE_NAME",
+        "HOLOHUB_IMAGE",
+        "HOLOHUB_ALWAYS_BUILD",
+    ]
+    for var in sorted(holohub_env_vars):
+        print(f"  {var}: {os.environ.get(var) or '(not set)'}")
+
+    print(f"\n{Color.blue('Holoscan Environment Variables:')}")
+    holoscan_env_vars = ["HOLOSCAN_SDK_VERSION", "HOLOSCAN_INPUT_PATH"]
+    for var in sorted(holoscan_env_vars):
+        print(f"  {var}: {os.environ.get(var) or '(not set)'}")
+
+    print(f"\n{Color.blue('Other Relevant Environment Variables:')}")
+    other_env_vars = [
+        "PYTHONPATH",
+        "PATH",
+        "LD_LIBRARY_PATH",
+        "CMAKE_BUILD_TYPE",
+        "DOCKER_BUILDKIT",
+        "XDG_SESSION_TYPE",
+        "XDG_RUNTIME_DIR",
+    ]
+    for var in sorted(other_env_vars):
+        print(f"  {var}: {os.environ.get(var) or '(not set)'}")
```

Comment:
`PYTHONPATH` here is accurate for the general environment. We also update `PYTHONPATH` at run command time in case the app build has generated a `holohub` module. From `./holohub run endoscopy_tool_tracking --language python --dryrun`:
```py
2025-06-18 13:09:00 [dryrun] $ export PYTHONPATH=/workspace/holohub/benchmarks/holoscan_flow_benchmarking:/opt/nvidia/holoscan/python/lib:/workspace/holohub:/opt/nvidia/holoscan/lib/../python/lib:/workspace/holohub/build/endoscopy_tool_tracking/python/lib:/workspace/holohub
```

What do you think about adding an optional project arg to print the anticipated environment when the project is actually run, i.e. `./holohub env-info endoscopy_tool_tracking`? Does not need to block this PR.

Reply:
yes, that's a good idea I think it could be done by another PR. we may consider providing info about the specific location of cmakefiles folder and docker inspect instructions as well.